fcli-ssc/src/main/java/com/fortify/cli/ssc/role/cli/mixin/SSCRoleResolverMixin.java

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * (c) Copyright 2020 Micro Focus or one of its affiliates
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a 
+ * copy of this software and associated documentation files (the 
+ * "Software"), to deal in the Software without restriction, including without 
+ * limitation the rights to use, copy, modify, merge, publish, distribute, 
+ * sublicense, and/or sell copies of the Software, and to permit persons to 
+ * whom the Software is furnished to do so, subject to the following 
+ * conditions:
+ * 
+ * The above copyright notice and this permission notice shall be included 
+ * in all copies or substantial portions of the Software.
+ * 
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY 
+ * KIND, EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE 
+ * WARRANTIES OF MERCHANTABILITY, FITNESS FOR A PARTICULAR 
+ * PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE 
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, 
+ * DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION OF 
+ * CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN 
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS 
+ * IN THE SOFTWARE.
+ ******************************************************************************/
+package com.fortify.cli.ssc.role.cli.mixin;
+
+import com.fortify.cli.ssc.role.helper.SSCRoleDescriptor;
+import com.fortify.cli.ssc.role.helper.SSCRoleHelper;
+import io.micronaut.core.annotation.ReflectiveAccess;
+import kong.unirest.UnirestInstance;
+import lombok.Getter;
+import lombok.SneakyThrows;
+import picocli.CommandLine.Option;
+import picocli.CommandLine.Parameters;
+
+@ReflectiveAccess
+public class SSCRoleResolverMixin {
+    
+    public static abstract class AbstractSSCRoleMixin {
+        public abstract String getRoleNameOrId();
+
+        @SneakyThrows
+        public SSCRoleDescriptor getRole(UnirestInstance unirestInstance){
+            return SSCRoleHelper.getRole(unirestInstance, getRoleNameOrId());
+        }
+        
+        public String getRoleId(UnirestInstance unirestInstance) {
+            return getRole(unirestInstance).getRoleId();
```

Comment:
Not sure how much data is in a role object and whether the endpoint supports the `fields` parameter, but you may want to change this to `return getRole(unirestInstance, "id").getRoleId();`, and update the `getRole` method to accept a `String... fields` parameter, like you have in the helper class. See `SSCAppVersionResolver` for example.

Reply:
Done.